sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java

```diff
@@ -2172,6 +2172,21 @@ public PipelineOptions getPipelineOptions() {
                           c.output(KV.of(key, output));
                         }
 
+                        @ProcessRetraction
+                        public void processRetraction(final ProcessContext c) {
+                          if (fn instanceof CombineFn) {
+                            c.outputRetraction(
+                                KV.of(
+                                    c.element().getKey(),
+                                    ((CombineFn<? super InputT, ?, OutputT>) fn)
+                                        .apply(c.element().getValue())));
+
+                          } else {
+                            throw new IllegalStateException(
+                                String.format("Unknown type of CombineFn: %s", fn.getClass()));
```

Comment:
`CombineFnWithContext` is not unknown - need to have a different error message for that saying it is unsupported.